x-pack/plugins/observability_solution/dataset_quality/public/state_machines/dataset_quality_controller/src/state_machine.ts

```diff
@@ -606,6 +681,29 @@ export const createDatasetQualityControllerStateMachine = ({
           end,
         });
       },
+
+      loadDegradedFieldsPerDataStream: (context) => {
+        if (!context.flyout.dataset || !context.flyout.insightsTimeRange) {
+          fetchDatasetDetailsFailedNotifier(toasts, new Error(noDatasetSelected));
```

Comment:
**suggestion**: if there is an error to handle, you can make this handling more responsible in the state machine service throwing the error and raising the an action for fetchDatasetDetailsFailedNotifier in the state machine definition.

So here the service should only trigger the error, and the `onError` handler when this service fails should raise the action.

Reply:
Discussed offline, will remove the toast